src/lazy/binary/raw/value.rs

```diff
@@ -78,20 +116,160 @@ impl<'top> LazyRawValue<'top, BinaryEncoding_1_0> for LazyRawBinaryValue_1_0<'to
+    /// Returns the input stream index range that contains the bytes representing the complete
+    /// annotations wrapper, including its opcode, wrapper length, annotations sequence length,
+    /// and the sequence itself.
+    pub fn range(&self) -> Range<usize> {
+        self.value.encoded_value.annotations_range().unwrap()
     }
 
-    fn span(&self) -> &[u8] {
+    /// Returns the encoded bytes representing the complete annotations wrapper, including its
+    /// opcode, wrapper length, annotations sequence length, and the sequence itself.
+    pub fn span(&self) -> Span<'top> {
         let range = self.range();
-        // Subtract the `offset()` of the ImmutableBuffer to get the local indexes for start/end
-        let local_range = (range.start - self.input.offset())..(range.end - self.input.offset());
-        &self.input.bytes()[local_range]
+        let start = range.start - self.value.input.offset();
+        let end = start + range.len();
+        let bytes = &self.value.input.bytes()[start..end];
+        Span::with_range(range, bytes)
+    }
+
+    /// Returns the input stream index range that contains the bytes representing the annotations
+    /// wrapper's opcode.
+    pub fn opcode_range(&self) -> Range<usize> {
+        let stream_start = self.range().start;
+        stream_start..stream_start + 1
+    }
+
+    /// Returns the encoded bytes representing the annotations wrapper's opcode.
+    pub fn opcode_span(&self) -> Span<'top> {
+        let stream_range = self.opcode_range();
+        let local_range = 0..1;
+        let bytes = &self.span().bytes()[local_range];
+        Span::with_range(stream_range, bytes)
+    }
+
+    /// Returns the encoded bytes representing the annotations wrapper's header (that is: the opcode,
+    /// wrapper length, and sequence length, but not the annotations sequence).
+    pub fn header_span(&self) -> Span<'top> {
+        let range = self.range();
+        let sequence_length = self.value.encoded_value.annotations_sequence_length as usize;
+        let local_end = range.len() - sequence_length;
+        let bytes = &self.span().bytes()[..local_end];
+        Span::with_range(range.start..range.start + local_end, bytes)
+    }
+
+    // TODO: separate span accessors for the wrapper length and sequence length?
+
+    /// Returns the encoded bytes representing the annotations wrapper's annotations sequence.
+    pub fn sequence_span(&self) -> Span<'top> {
+        let range = self.range();
+        let sequence_length = self.value.encoded_value.annotations_sequence_length as usize;
+        let local_start = range.len() - sequence_length;
+        let bytes = &self.span().bytes()[local_start..];
+        let stream_start = range.start + local_start;
+        Span::with_range(stream_start..stream_start + sequence_length, bytes)
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct EncodedBinaryValueData_1_0<'a, 'top> {
+    value: &'a LazyRawBinaryValue_1_0<'top>,
+}
```

Comment:
🗺️ This type offers methods for accessing the various encoding components of a binary 1.0 value.